picobox/_stack.py

```diff
@@ -30,6 +30,32 @@ def __getattribute__(self, name):
 
 
 class push:
+    """Context manager to push a :class:`Box` instance to the top of the stack.
+
+    The box on the top is used by :func:`put`, :func:`get` and :func:`pass_`
+    functions (not methods) and together they define a so called Picobox's
+    stacked interface. The idea behind stacked interface is to provide a way
+    to easily switch DI containers (boxes) without changing injections.
+
+    Here's a minimal example of how push can be used::
+
+        import picobox
+
+        @picobox.pass_('magic')
+        def do(magic):
+            return magic + 1
+
+        foobox = picobox.Box()
+        foobox.put('magic', 42)
+
+        barbox = picobox.Box()
+        barbox.put('magic', 13)
+
+        with picobox.push(foobox):
+            with picobox.push(barbox):
+                assert do() == 13
```

Comment:
As I understand the example using `barbox` box should result to `14` after executing `do()`.

Reply:
Damn you right! :)